### DISCUSSION TITLE
Why `HTTPException` gets sent as exception?

### DISCUSSION THREAD

**Discussion by Kludex** [getsentry/sentry-python#2494] [Q&A]
Why `starlette.exceptions.HTTPException` and `fastapi.exceptions.HTTPException` is sent as event to sentry?

**Comment by smeubank** [comment]
heyyyy @Kludex 👋 

do you have a specific example? It's turned on with the integrations by default, we will have to check if this is only happening with Starlette and FastAPI integrations. Checking some older discussion, potentially related. It seems this is not always the case. Are the exceptions associated with any specific status code? Can you provide an example event that is sent, and the init setup you have for that python app?

https://github.com/getsentry/sentry-python/issues/642

**Reply by Kludex** [reply]
Hey Steven! 👋 😁

Yeah:

```python
from fastapi import HTTPException, FastAPI

app = FastAPI()

@app.get('/')
async def health():
    raise HTTPException(status_code=502, detail='potato')
```
I'm just trying to understand why `HTTPException` is not whitelisted.

**Reply by smeubank** [reply]
this does not generate an error event for sentry? [Health checks](https://docs.sentry.io/product/data-management-settings/filtering/#transactions-coming-from-healthcheck) are filtered out but that should only be for transactions AFAIK

I will forward this to the team, and see if anyone has a better idea.

**Reply by Kludex** [reply]
I have this enabled:

<img width="621" alt="Screenshot 2024-01-23 at 11 28 54" src="https://github.com/getsentry/sentry-python/assets/7353520/d6ee0edb-42e5-4098-8f0f-d0e4e9826b6a">

And the transaction is called `/health`:

<img width="163" alt="Screenshot 2024-01-23 at 11 29 29" src="https://github.com/getsentry/sentry-python/assets/7353520/6e91a22d-b0c5-4f74-9fef-a12814fa8d2d">

What is missing for it to be filtered?

**Comment by sentrivana** [comment]
Hey @Kludex 👋🏻 We don't capture all `HTTPException`s, only [those with a 500+ status code](https://github.com/getsentry/sentry-python/blob/c8154be61eb473d954db2a998b4647a64065a73e/sentry_sdk/integrations/starlette.py#L212-L218). This is a pattern in most web framework integrations we have since these 500+ `HTTPException`s usually wrap unhandled errors and/or are important to know about.

**Reply by sentrivana** [reply]
If anyone wants to fine-tune this, [`before_send`](https://docs.sentry.io/platforms/python/configuration/filtering/#using-platformidentifier-namebefore-send-) can be used to drop unwanted events before they're sent to Sentry.

**Comment by sentrivana** [comment]
We've just shipped an improvement in https://github.com/getsentry/sentry-python/releases/tag/2.5.0 that enables you to pick which HTTP status codes to report to Sentry. Check out the `failed_request_status_code` option in the [docs](https://docs.sentry.io/platforms/python/integrations/fastapi/#options).

**Reply by Kludex** [reply]
Does it help here, tho? I want to see 500s, but not when I raise with `HTTPException`...

**Reply by sentrivana** [reply]
Gotcha now @Kludex -- no in that case it doesn't help. You can still drop unwanted events in a custom [before_send](https://docs.sentry.io/platforms/python/configuration/options/#before-send) though.
